src/core/utils/request/urlFromRequestOrCache.ts

```diff
@@ -0,0 +1,17 @@
+const urlCache = new Map<string, URL>()
```

Comment:
Why not `WeakMap`?

Reply:
Mostly to keep URL objects between requests, likely a slight over optimization, which keeps subsequent requests slightly more efficient at the expense of memory. 

Both should be generally helpful